src/data/fetchProfessor.ts

```diff
@@ -0,0 +1,52 @@
+
+import fetch from "node-fetch";
+import type { fetchProfessor_params, professorData } from "./nebulaInterfaces.js";
+import { unRegister } from "./removeLater.js";
+
+// TO RUN, RUN THIS WITH 
+// ts-node-esm fetchProfessor.ts
+
+let NEBULA_API_KEY = "EM~eW}G<}4qx41fp{H=I]OZ5MF6T:1x{<GF:~v<";
+
+// Helper function to fetch data from Nebula API. Returns a promise.
+function fetchHelper(paramsObj: fetchProfessor_params) {
+    const headers = {
+        "x-api-key": unRegister(NEBULA_API_KEY),
+        Accept: "application/json",
+    };
+    const getDataPromise = new Promise(async (resolve, reject) => {
+        try {
+            const res = await fetch(
+                `https://api.utdnebula.com/professor?first_name=${paramsObj.firstName}&last_name=${paramsObj.lastName}`,
+                {
+                    method: "GET",
+                    headers: headers,
+                }
+            );
+            resolve(res.json());
+        } catch (err) {
+            reject(new Error(err));
+        }
+    });
+
+    return getDataPromise;
+}
```

Comment:
This can be made into one functions that handles errors, and then we can remove getNebulaProfessor. Also once we have these shorter functions and since we share fetchOptions, we can put them in one **fetch.ts** file

Example:

```js
async function fetchNebulaCourse(paramsObj: fetchCourse_params): Promise<courseData> | null {
    try {
        const res = await fetch(`https://api.utdnebula.com/course?course_number=${paramsObj.courseNumber}&subject_prefix=${paramsObj.subjectPrefix}`, fetchOptions);
        const json = await res.json();
        if (json.data == null) throw new Error("Null data");
        const data: courseData = json.data[0];
        return data;
    } catch (error) {
        return null;
    }
}
```